webapp/channels/src/components/user_settings/display/render_emoticons_as_emoji/render_emoticons_as_emoji.tsx

```diff
@@ -0,0 +1,167 @@
+
+    const [isSaving] = useState<boolean>(false);
+
+    const [serverError] = useState<string | undefined>(undefined);
+
+    const minRef = useRef<SettingItemMinComponent>(null);
+
+    const focusEditButton = () => {
+        minRef.current?.focus();
+    };
+
+    useEffect(() => {
+        if (active && areAllSectionsInactive) {
+            focusEditButton();
+        }
+    }, [active, areAllSectionsInactive]);
+
+    const handleOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setRenderEmoticonsAsEmojiState(e.currentTarget.value);
+        a11yFocus(e.currentTarget);
+    };
+
+    const handleUpdateSection = (section: string) => {
+        const renderEmoticonsAsEmojiPreference = {
+            user_id: userId,
+            category: Preferences.CATEGORY_DISPLAY_SETTINGS,
+            name: Preferences.RENDER_EMOTICONS_AS_EMOJI,
+            value: renderEmoticonsAsEmojiState,
+        };
+        if (!section) {
+            actions.savePreferences(userId, [renderEmoticonsAsEmojiPreference]);
+        }
+        updateSection(section);
+    };
+
+    const handleSubmit = () => {
+        const renderEmoticonsAsEmojiPreference = {
+            user_id: userId,
+            category: Preferences.CATEGORY_DISPLAY_SETTINGS,
+            name: Preferences.RENDER_EMOTICONS_AS_EMOJI,
+            value: renderEmoticonsAsEmojiState,
+        };
+        actions.savePreferences(userId, [renderEmoticonsAsEmojiPreference]);
+        updateSection('');
+    };
+
+    if (active) {
+        return (
+            <SettingItemMax
+                title={
+                    <FormattedMessage
+                        id='user.settings.display.renderEmoticonsAsEmojiTitle'
+                        defaultMessage='Auto-render emoticons as emojis'
+                    />
+                }
+                inputs={[
+                    <fieldset key='renderEmoticonsAsEmojiSetting'>
+                        <legend className='form-legend hidden-label'>
+                            <FormattedMessage
+                                id='user.settings.display.renderEmoticonsAsEmojiTitle'
+                                defaultMessage='Auto-render emoticons as emojis'
+                            />
+                        </legend>
+                        <div className='radio'>
+                            <label>
+                                <input
+                                    id='renderEmoticonsAsEmojiOn'
+                                    type='radio'
+                                    value='true'
+                                    name='renderEmoticonsAsEmoji'
+                                    checked={renderEmoticonsAsEmojiState === 'true'}
+                                    onChange={handleOnChange}
+                                />
+                                <FormattedMessage
+                                    id='user.settings.display.renderEmoticonsAsEmojiOn'
```

Comment:
I can't remember why we started using different IDs for "on/off" for each of these settings, but I think we should stop adding more of those. The ID of this one isn't quite correct, but we seem to use it in a few other places, so we can use it here as well
```suggestion
                                    id='user.settings.advance.on'
```

You'll also need to remember to update `en.json` again as well